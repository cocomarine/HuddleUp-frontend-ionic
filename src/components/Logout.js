import React from "react";
import { useNavigate } from "react-router-dom";
import { useAuthContext } from "../hooks/useAuthContext";
// import "../styles/logout.css";

const Logout = () => {
  const { dispatch } = useAuthContext();

  const navigate = useNavigate();

  const changeLocation = (redirect) => {
    navigate(redirect, { replace: true });
    // window.location.reload();
  };

  const handleLogout = () => {
    localStorage.removeItem("user");
    dispatch({ type: "LOGOUT "});
    changeLocation("/");
  };

  return (
    <div className="logoutt">
      <button type="submit" onClick={handleLogout}>
        Logout
      </button>
    </div>
  );
};

export default Logout;
