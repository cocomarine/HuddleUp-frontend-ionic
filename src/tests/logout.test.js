import React from "react";
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Logout from "../components/Logout";
import { AuthContextProvider } from "../contexts/AuthContext";

beforeEach(() => {
  render(
    <BrowserRouter>
      <AuthContextProvider>
        <Logout />
      </AuthContextProvider>
    </BrowserRouter>
  );
});

afterEach(cleanup);

describe("Logout", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <AuthContextProvider>
        <Logout />
      </AuthContextProvider>
    </BrowserRouter>
  );
  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  xit("when clicked, takes you to login page", () => {});
});
