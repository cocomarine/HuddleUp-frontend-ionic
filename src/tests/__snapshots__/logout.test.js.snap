// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders correctly 1`] = `
<DocumentFragment>
  <div
    class="logoutt"
  >
    <button
      type="submit"
    >
      Logout
    </button>
  </div>
</DocumentFragment>
`;
